akka-actor/src/main/scala/akka/dispatch/affinity/AffinityPool.scala

```diff
@@ -168,23 +250,21 @@ private[akka] class AffinityPool(
       else awaitTermination(terminationCondition.awaitNanos(nanos))
     }
 
-    locked(bookKeepingLock) {
+    locked {
       // need to hold the lock to avoid monitor exception
       awaitTermination(unit.toNanos(timeout))
     }
-
   }
 
-  private def attemptPoolTermination() =
-    locked(bookKeepingLock) {
```

Comment:
This was only ever called while holding the lock, so re-locking was superfluous.